src/OrleansRuntime/Scheduler/InvokeWorkItem.cs

```diff
@@ -43,7 +43,8 @@ public override void Execute()
             try
             {
                 var grain = activation.GrainInstance;
-                var runtimeClient = (ISiloRuntimeClient)grain.GrainReference.RuntimeClient;
+                var runtime = (GrainReferenceRuntime)grain.GrainReference.Runtime;
```

Comment:
Yes, this one annoyed me too. But since it was already downcasting before, I thought not to improve it in this PR. If you prefer, I could add the IRuntimeClient directly in ActivationData